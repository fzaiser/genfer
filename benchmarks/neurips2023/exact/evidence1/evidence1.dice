let evidence = flip 0.5 in
let coin  = flip 0.5 in
if evidence then
    let tmp = observe coin in evidence
    else evidence
