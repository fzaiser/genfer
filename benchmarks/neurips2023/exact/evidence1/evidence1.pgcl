nat evidence;
nat coin;

evidence := bernoulli(1/2);
coin := bernoulli(1/2);
if (evidence = 1) {
    observe(coin = 1);
} else {skip}

?Pr[evidence];
