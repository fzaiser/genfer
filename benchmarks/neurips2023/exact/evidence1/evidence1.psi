// flags: --dp
def main(){
    evidence := flip(1/2);
    if evidence {
        coin := flip(1/2);
        observe(coin);
    }
    return evidence; // expected: 1/3·δ(1)[evidence]+2/3·δ(0)[evidence]
}
