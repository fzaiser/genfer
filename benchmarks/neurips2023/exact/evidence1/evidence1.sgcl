evidence ~ Bernoulli(0.5);
coin ~ Bernoulli(0.5);
if evidence = 1 {
    observe coin = 1;
}
return evidence;
