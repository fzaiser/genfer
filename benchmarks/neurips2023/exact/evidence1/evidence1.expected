0.33333333
p(1) / Z = 1/3
1/3·δ(1)[evidence]
1/3*evidence
